### DISCUSSION TITLE
Separate Form and List and show them on differnt pages

### DISCUSSION THREAD

**Discussion by AnH13-6** [neuefische/questions#237] [Web] [answered]
Hey, I would like to separate my entry form from my book list and show them on different pages in my app. As the map-functions need information from the form I don't know how to separate the code that currently is in the same data. 

Here is the Link to my repo: 

https://github.com/AnH13-6/capstone-project-two/compare/styling?expand=1

Thanks in advance!

**Comment by mbosselmann** [comment, marked as answer]
Anne and I worked on a solution where we moved the books state into the `_app.js` file:

```js
import {useState, useEffect} from 'react';

import {GlobalStyle} from '../styles';

export default function App({Component, pageProps}) {
	const [books, setBooks] = useState([]);

	useEffect(() => {
		fetchBooks();
		async function fetchBooks() {
			try {
				const response = await fetch('/api/books');
				if (response.ok) {
					const data = await response.json();
					setBooks(data);
				} else {
					throw new Error('fetch failed');
				}
			} catch (error) {
				console.log(error);
			}
		}
	}, []);

	return (
		<>
			<GlobalStyle />
			<Component {...pageProps} books={books} onHandleBooks={setBooks} />
		</>
	);
}
```

We divided the `BookList` component into two components (one for the list and one for the form), changed the pages accordingly and passed the props through the pages to the relevant components.
